pallets/nimbus-async-backing/src/lib.rs

```diff
@@ -0,0 +1,96 @@
+// Copyright Moonsong Labs
+// This file is part of Moonkit.
+
+// Moonkit is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonkit is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonkit.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Pallet that allows block authors to include their identity in a block via an inherent.
+//! Currently the author does not _prove_ their identity, just states it. So it should not be used,
+//! for things like equivocation slashing that require authenticated authorship information.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub mod consensus_hook;
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+pub use pallet::*;
+
+use frame_support::pallet_prelude::*;
+use sp_consensus_slots::Slot;
+
+/// The InherentIdentifier for nimbus's extension inherent
+pub const INHERENT_IDENTIFIER: InherentIdentifier = *b"nimb-ext";
+
+/// Parachain slot implementation
+/// If you don't need to have slots at parachain level, you can use the `RelaySlot` implementation.
+pub trait ParachainSlot {
+	/// Get the current slot
+	fn get_current_slot() -> Option<Slot>;
+	/// Compute parachain slot from relay chain timestamp
+	fn para_slot_from_relay_timestamp(relay_chain_timestamp: u64) -> Option<Slot>;
+}
+
+/// Parachain slot implementation that use the relay chain slot directly
+pub struct RelaySlot;
+impl ParachainSlot for RelaySlot {
+	fn get_current_slot() -> Option<Slot> {
+		None
+	}
+	fn para_slot_from_relay_timestamp(_: u64) -> Option<Slot> {
+		None
+	}
+}
+
+/*/// Fixed time slot
+struct FixedSlot<GetSlotDuration>(PhantomData<GetSlotDuration>);
+
+impl<GetSlotDuration: Get<u64>> ParachainSlot for FixedSlot<GetSlotDuration> {
```

Comment:
I think this is a good idea to provide an implementation for parachains that have a slot duration set. But why is it commented? Are you planning to remove it?

Reply:
This code should be on tanssi side because it need aura, I will cleanup the comments